### DISCUSSION TITLE
Disable Hetzner LoadBalancer

### DISCUSSION THREAD

**Discussion by eiskasten** [mysticaltech/terraform-hcloud-kube-hetzner#740] [Q&A] [answered]
Am I correct, that it is currently not possible to deploy a cluster with the terraform script without deploying a Hetzner loadbalancer? In my case the ingress would be sufficient for loadbalancing tasks. My workloads do not require a dedicated loadbalancer. Am I missing the benefit using a dedicated loadbalancer for a single node cluster?

**Comment by schlichtanders** [comment]
I am also surprised by this. I am using kubernetes ingress and am unsure what the benefit is of paying an extra hetzner loadbalancer.

**Reply by schlichtanders** [reply, marked as answer]
I found two options in the `kube.tf` so far:
- `enable_klipper_metal_lb = "true"`, this seems to be deployed within the kubernetes itself - no extra hetzner cost are needed. Downside: new pods will also be scheduled on the control plane nodes (see [this discussion](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/discussions/201) which is actually not answered) 
- `use_control_plane_lb = true`, don't know why this is disabled by default and also don't know what this is doing. Anyone any idea?

**Reply by eiskasten** [reply]
> I found two options in the `kube.tf` so far:
> 
>     * `enable_klipper_metal_lb = "true"`, this seems to be deployed within the kubernetes itself - no extra hetzner cost are needed. Downside: new pods will also be scheduled on the control plane nodes (see [this discussion](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/discussions/201) which is actually not answered)

Ok, this is actually the settings I was looking for. I wanted to schedule pods on control plane nodes anyway. I am sorry for that!

However, since you seem to be right and it seems not to be possible to omit the loadbalancer without scheduling on the control plane, I think this discussion should be continued.

> 
>     * `use_control_plane_lb = true`, don't know why this is disabled by default and also don't know what this is doing. Anyone any idea?

To be honest, I am not 100% sure what exactly it does. However, I think it is only related to HA mode. The corresponding PR is #256, maybe this does help you a bit?

**Reply by schlichtanders** [reply]
I found another discussion about the use of `use_control_plane_lb` : https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/discussions/302

it seems that this one is needed if you require highly available access to kubenertes API (which I actually need)

**Reply by eiskasten** [reply]
Ok, so that is intended for what I was thinking about.
